metadata-ingestion/tests/unit/sdk_v2/test_client_v2.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import Mock
+
+import pytest
+
+from datahub.errors import ItemNotFoundError, MultipleItemsFoundError, SdkUsageError
+from datahub.ingestion.graph.client import DataHubGraph
+from datahub.ingestion.graph.config import DatahubClientConfig
+from datahub.metadata.urns import CorpUserUrn
+from datahub.sdk.main_client import DataHubClient
+
+
+@pytest.fixture
+def mock_graph() -> Mock:
+    graph = Mock(spec=DataHubGraph)
+    graph.exists.return_value = False
+    return graph
+
+
+def test_client_creation(mock_graph: Mock) -> None:
+    assert DataHubClient(graph=mock_graph)
+    assert DataHubClient(server="https://example.com", token="token")
+
+
+def test_client_init_errors(mock_graph: Mock) -> None:
+    config = DatahubClientConfig(server="https://example.com", token="token")
+
+    with pytest.raises(SdkUsageError):
+        DataHubClient(server="https://example.com", graph=mock_graph)  # type: ignore
+    with pytest.raises(SdkUsageError):
+        DataHubClient(server="https://example.com", config=config)  # type: ignore
+    with pytest.raises(SdkUsageError):
+        DataHubClient(config=config, graph=mock_graph)  # type: ignore
+    with pytest.raises(SdkUsageError):
+        DataHubClient()  # type: ignore
+
+
+def test_resolve_user(mock_graph: Mock) -> None:
+    client = DataHubClient(graph=mock_graph)
+
+    # This test doesn't really validate the graphql query or vars.
+    # It probably makes more sense to test via smoke-tests.
+
+    mock_graph.get_urns_by_filter.return_value = []
+    with pytest.raises(ItemNotFoundError):
+        client.resolve.user(name="User")
+
+    mock_graph.get_urns_by_filter.return_value = ["urn:li:corpuser:user"]
+    assert client.resolve.user(name="User") == CorpUserUrn("urn:li:corpuser:user")
```

Comment:
is this the expected usage?
to fetch the URN so later we can fetch other data for that URN? 
if so, this is promoting a two-round-trip for any data fetching for something that could be done in a single round 🤔

Reply:
this is purely for _resolving_ which can be useful for e.g. populating the ownership field. 

```
dataset.add_owner(client.resolve.user(email=...))
```

I don't expect people to be using this method in isolation - that's only for testing.